saas_demo/models/saas_demo.py

```diff
@@ -135,8 +135,11 @@ def filter_free_operators(op):
         operators.write({
             'update_repos_state': 'updating',
         })
-        # close transaction to make update_repos_state update visible
-        self.env.cr.commit()
+        test_module = self.env['ir.module.module'].search([('name', '=', 'saas_demo_test')])
+        # no need to commit with the installed test module
```

Comment:
You comment what are you doing, but not why.
>         # no need to commit with the installed test module


```suggestion
        # There is no clean way to determine whether we are on tests or not.
        # So, assume that we are on tests if modules is installed
        
```